include/fc/variant.hpp

```diff
@@ -654,6 +661,36 @@ namespace fc
          c.insert( item.as<T>( max_depth - 1 ) );
    }
 
+   template<size_t N>
+   void to_variant( const std::array<char,N>& bi, variant& v, uint32_t max_depth = 1 )
+   {
+      v = variant( to_hex( bi.data(), N ) );
+   }
+   template<size_t N>
+   void from_variant( const variant& v, std::array<char,N>& bi, uint32_t max_depth = 1 )
+   {
+      std::string ve = v.as_string();
+      if( ve.size() )
+         from_hex( ve, bi.data(), std::min<size_t>( ve.size() / 2, bi.size() ) );
+      else
+         memset( bi.data(), 0, bi.size() );
```

Comment:
Calling `memset` unconditionally is perhaps more readable, but it also means we'll typically write all of `bi` twice, which makes the call twice as expensive. Thus it could also be that, if `ve` is defined but small, we call `from_hex` then `memset` the leftover bytes.

Reply:
Agree in principle, but I tried to maintain existing functionality. See @abitmore 's comment: https://github.com/bitshares/bitshares-fc/pull/137#discussion_r291979962